catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/UpdateParent.java

```diff
@@ -0,0 +1,153 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+
+/**
+ * Update the parent metacard by calling an {@link UpdateField} and then by calling the
+ * catalog framework.
+ */
+public class UpdateParent implements FindChildrenStreamEndPlugin.Handler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdateParent.class);
+
+    private static final long INITIAL_RETY_MILLISECONDS = TimeUnit.SECONDS.toMillis(1);
+
+    private static final long MAX_RETRY_MILLISECONDS = TimeUnit.MINUTES.toMillis(5);
+
+    private final UpdateField updateField;
+
+    private CatalogUpdateRetry catalogUpdateRetry = new CatalogUpdateRetry();
+
+    public UpdateParent(UpdateField updateField) {
+        this.updateField = updateField;
+    }
+
+    public void setCatalogUpdateRetry(CatalogUpdateRetry catalogUpdateRetry) {
+        this.catalogUpdateRetry = catalogUpdateRetry;
+    }
+
+    @Override
+    public void handle(Context context, Metacard parent, List<Metacard> children) {
+        updateField.updateField(parent, children, context);
+    }
+
+    @Override
+    public void end(Context context, Metacard parentMetacard) {
+
+        updateField.end(parentMetacard, context);
+
+        UpdateRequest updateRequest = createUpdateRequest(parentMetacard);
+
+        update(context, parentMetacard, updateRequest);
+
+    }
+
+    private void update(Context context, Metacard parentMetacard, UpdateRequest updateRequest) {
+        try {
+            catalogUpdateRetry.submitUpdateRequestWithRetry(context.getUdpStreamProcessor()
+                            .getCatalogFramework(),
+                    updateRequest,
+                    context.getUdpStreamProcessor()
+                            .getMetacardUpdateInitialDelay(),
+                    INITIAL_RETY_MILLISECONDS,
+                    MAX_RETRY_MILLISECONDS,
+                    update -> {
+                        LOGGER.debug("updated parent metacard: newMetacard={}",
+                                update.getNewMetacard()
+                                        .getId());
+                        context.setParentMetacard(update.getNewMetacard());
+                    });
+        } catch (RolloverActionException e) {
+            LOGGER.debug("unable to update parent metacard: metacardId={}",
+                    parentMetacard.getId(),
+                    e);
+        }
+    }
+
+    private UpdateRequest createUpdateRequest(Metacard parentMetacard) {
+        return new UpdateRequestImpl(parentMetacard.getId(), parentMetacard);
+    }
+
+    /**
+     * Update a field in the parent metacard. {@link #updateField} will be called with batches
+     * of child metacards. After all the batches have been submitted, then {@link #end(Metacard,Context)}
+     * will be called.
+     */
+    interface UpdateField {
```

Comment:
❓ Instead of embedding this interface and the `BaseUpdateField` abstract base class, should we consider making them top-level and putting them in their own package? I was looking at the `mpegts.plugins` package and it's getting big so this could be a good time to create logical sub-packages and reduce the number of inner interfaces and classes. Thoughts?